providers/azure.go

```diff
@@ -209,6 +209,13 @@ func (p *AzureProvider) EnrichSession(ctx context.Context, session *sessions.Ses
 		session.Email = email
 	}
 
+	user, err := p.getUser(ctx, session.AccessToken)
```

Comment:
Should we not check that the user field is not already set? Like we do for email?

Reply:
@JoelSpeed  please check this PR
https://github.com/oauth2-proxy/oauth2-proxy/pull/2535
it's based on this PR + small refactoring